readnode-primitives/src/lib.rs

```diff
@@ -88,8 +77,10 @@ impl CollectingTransactionDetails {
 
     pub fn final_status(&self) -> Option<views::FinalExecutionStatus> {
```

Comment:
Today, we tried to remember the logic of what we are doing here and why. How about write it down here to speed this process up in the future?